src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -894,6 +903,10 @@ internal virtual void PrepareSession()
                 // Store the UserAgent string
                 WebSession.UserAgent = UserAgent;
             }
+            if (MyInvocation.BoundParameters.ContainsKey("NoProxy"))
```

Comment:
```suggestion
            if (NoProxy.IsPresent)
```

Reply:
Maybe you could even remove the if and leave `WebSession.NoProxy = NoProxy.IsPresent;`

Reply:
This one is tricky since it's a switch, the -NoProxy flag will have been given when creating the WebSession, but if it's not present on subsequent requests do we assume they just don't feel the need to repeat the -NoProxy because they're using a session, or do we assume they no longer want the proxy. 

I was using presence of the switch to detect that but on reflection I think it's still not going to do that right. Perhaps proxy related settings should just be ignored on subsequent requests when a websession has been requested?

If we identified some of the items that we won't support changing on the fly in a session we could potentially simplify the amount of state being monitored for change.

Reply:
Without that presence check, this script will cause a reconnect on the first loop through using the websession (similarly for NoProxy and SslProtocol args):

```powershell
Measure-Command {
     Write-Host (Invoke-WebRequest https://postman-echo.com/get?i=1 -SessionVariable Session -SslProtocol Tls12 -NoProxy).StatusCode
    1..10 | ForEach-Object {
         write-host (Invoke-WebRequest https://postman-echo.com/get?i=$_ -WebSession $Session).StatusCode
     }
 }
```
With the check in, the only way to force it to see the change to the -NoProxy flag is to provide it explicitly set to $false

```powershell
Measure-Command {
     Write-Host (Invoke-WebRequest https://postman-echo.com/get?i=1 -SessionVariable Session -SslProtocol Tls12 -NoProxy).StatusCode
    1..10 | ForEach-Object {
         write-host (Invoke-WebRequest https://postman-echo.com/get?i=$_ -WebSession $Session -NoProxy:$false).StatusCode
     }
 }
```

Reply:
I don't know if my idea is correct, but what if once you set -NoProxy on the first request it stays NoProxy=true until you use -Proxy in a subsequent request? (And vice versa)

Reply:
I haven't thought about that before, but I have a suspicion it would make it even more complex.

Reply:
See latest commit(s) as I have implemented this idea.